### ISSUE TITLE
Production Ready

### ISSUE THREAD

**Issue by TimoLins** [vsivsi/meteor-file-collection#60]
Since I can't find any information on this: Is this package ready (and secure enough) for production?

EDIT: I really need this, because I think CollectionFS isn't ready for production yet. This package is my only hope.

**Comment by vsivsi** [comment]
Hi, Thanks for the question. Unfortunately I suspect there's no easy direct answer to your question... but I can say this:

The project is "stable" (and has been for many months) in that there are no major missing features or in-process changes scheduled for the codebase. I'm using it for my own work, as are many others.

I wrote this package, in part, because CollectionFS was much more complicated than I needed, and not as efficient as I wanted, but mostly because it didn't seem to be converging stability-wise (I was a contributor to the gridFS support in CollectionFS, and leveraged experience gained there to write this package). As I wrote in the [README](https://github.com/vsivsi/meteor-file-collection#design-philosophy), this package was intentionally designed to be simple, and simpler systems are easier to debug, understand and audit for potential issues.

As for security... This package uses the Meteor authentication, method invocation and pub/sub mechanisms, and so it can only be as secure as those are. And it is subject to the same caveats, namely that authentication tokens (which can be used to impersonate a user) are transmitted in the clear as either a cookie or HTTP header. For this reason, it is strongly advised that you use HTTPS/SSL for all connections traversing untrusted networks to protect the authentication exchange and subsequent token passing. In reality, Meteor can only be secure from MitM attacks via use of SSL anyway, because such an attacker could otherwise modify the Meteor and application code loaded onto the client to all sorts of bad things...
 
Anyway, there have been no security related issues reported in the ~1 year since the project started, which means that either I did an awesome job; or alternatively, that no one has really tried super hard to exploit it yet. Part of the beauty of the design of this package is that it adds very little in the way of security sensitive code to what Meteor already provides, so there's not too much to go wrong. If I had to guess where any likely security bugs will eventually appear, it will be in DoS-type exploits and not unauthorized access bugs. Anyway with that said, I'm sure there will eventually be bugs, because there always are...

So circling back around, I think that only *you* can answer if this package is "production ready" through your own due-diligence, given the stability and security requirements of your envisioned project. Because, well, read the [LICENSE](https://github.com/vsivsi/meteor-file-collection/blob/master/LICENSE).

I know that is probably a lot less satisfying than "Yes it's production ready" would be, but such a statement would imply a commitment on my part that I can't make given that I don't know your requirements and you won't be paying me to make good on any such commitment.
